core/domain/state_domain.py

```diff
@@ -18,6 +18,7 @@
 
 import copy
 import logging
+import re
```

Comment:
Can you please check how we import builtin libraries (I think we do this on top but I'm not sure!)